internal/services/apimanagement/api_management_resource.go

```diff
@@ -1188,18 +1180,19 @@ func resourceApiManagementServiceUpdate(d *pluginsdk.ResourceData, meta interfac
 		}
 
 		policyServiceId := policy.NewServiceID(subscriptionId, id.ResourceGroupName, id.ServiceName)
-		// remove the existing policy
-		if delResp, err := policyClient.Delete(ctx, policyServiceId, policy.DeleteOperationOptions{}); err != nil {
-			if !response.WasNotFound(delResp.HttpResponse) {
-				return fmt.Errorf("removing Policies from %s: %+v", id, err)
-			}
-		}
-
-		// then add the new one, if it exists
 		if policyContract != nil {
 			if _, err := policyClient.CreateOrUpdate(ctx, policyServiceId, *policyContract, policy.CreateOrUpdateOperationOptions{}); err != nil {
 				return fmt.Errorf(" setting Policies for %s: %+v", id, err)
 			}
+		} else {
+			oldPolicy, newPolicy := d.GetChange("policy")
+			if len(oldPolicy.([]interface{})) > 0 && len(newPolicy.([]interface{})) == 0 {
+				if resp, err := policyClient.Delete(ctx, policyServiceId, policy.DeleteOperationOptions{}); err != nil {
+					if !response.WasNotFound(resp.HttpResponse) {
+						return fmt.Errorf("removing Policies from %s: %+v", id, err)
+					}
+				}
+			}
```

Comment:
Rather than trying to determine whether the `policy` list has changed from >zero to zero elements, we should just check whether the current length is zero. Users can use `ignore_changes` if they do not wish to manage this property.

Reply:
Thanks for your time, the code has been updated, could you please take a look again?

Reply:
Hi @manicminer I removed the conditional policy deletion in the update func because I assume that if `d.HasChange("policy")` and `policyContract == nil`, there is only one possibility, which is to change the computed + optional property `policy` from being configured to being empty(`policy = []` ) in TF configuration. So I assume that there is no need to add a conditional deletion here, any feedback feel free let me know, thank you.